samples/bluetooth/iso_connected_benchmark/src/main.c

```diff
@@ -0,0 +1,1042 @@
+{
+	char role_char;
+	const char central_char = 'c';
+	const char peripheral_char = 'p';
+	const char quit_char = 'q';
+
+	while (true) {
+		printk("Choose device role - type %c (central role) or %c (peripheral role), or %c to quit: ",
+			central_char, peripheral_char, quit_char);
+
+		role_char = console_getchar();
+
+		printk("%c\n", role_char);
+
+		if (role_char == central_char) {
+			printk("Central role\n");
+			return ROLE_CENTRAL;
+		} else if (role_char == peripheral_char) {
+			printk("Peripheral role\n");
+			return ROLE_PERIPHERAL;
+		} else if (role_char == quit_char) {
+			printk("Quitting\n");
+			return ROLE_QUIT;
+		} else if (role_char == '\n' || role_char == '\r') {
+			continue;
+		}
+
+		printk("Invalid role: %c\n", role_char);
+	}
+}
+
+static void print_stats(char *name, struct iso_recv_stats *stats)
+{
+	uint32_t total_packets;
+
+	total_packets = stats->iso_recv_count + stats->iso_lost_count;
+
+	LOG_INF("%s: Received %u/%u (%.2f%%) - Total packets lost %u",
+		name, stats->iso_recv_count, total_packets,
+		(float)stats->iso_recv_count * 100 / total_packets,
+		stats->iso_lost_count);
+
+}
+
+static void iso_timer_timeout(struct k_work *work)
+{
+	int ret;
+	struct net_buf *buf;
+
+	/* Reschedule as early as possible to reduce time skewing
+	 * Use the ISO interval minus a few microseconds to keep the buffer
+	 * full. This might occasionally skip a transmit, i.e. where the host
+	 * calls `bt_iso_chan_send` but the controller only sending a single
+	 * ISO packet.
+	 */
+	k_work_reschedule(&iso_send_work, K_USEC(iso_tx_qos.interval - 100));
+
+	for (int i = 0; i < cis_create_count; i++) {
+		buf = net_buf_alloc(&tx_pool, K_FOREVER);
+		if (buf == NULL) {
+			LOG_ERR("Could not allocate buffer");
+			return;
+		}
+
+		net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+		net_buf_add_mem(buf, iso_data, iso_tx_qos.sdu);
+		ret = bt_iso_chan_send(&iso_chans[i], buf);
+		if (ret < 0) {
+			LOG_ERR("Unable to send data: %d", ret);
+			net_buf_unref(buf);
+			break;
+		}
+		iso_send_count++;
+
+		if ((iso_send_count % 100) == 0) {
+			LOG_INF("Sending value %u", iso_send_count);
+		}
+	}
+}
+
+static void iso_recv(struct bt_iso_chan *chan,
+		     const struct bt_iso_recv_info *info,
+		     struct net_buf *buf)
+{
+	uint32_t total_packets;
+	static bool stats_latest_arr[1000];
+	static size_t stats_latest_arr_pos;
+
+	/* NOTE: The packets received may be on different BISes */
+
+	if (info->flags == BT_ISO_FLAGS_VALID) {
+		stats_current_conn.iso_recv_count++;
+		stats_overall.iso_recv_count++;
+		stats_latest_arr[stats_latest_arr_pos++] = true;
+	} else {
+		stats_current_conn.iso_lost_count++;
+		stats_overall.iso_lost_count++;
+		stats_latest_arr[stats_latest_arr_pos++] = false;
+	}
+
+	if (stats_latest_arr_pos == sizeof(stats_latest_arr)) {
+		stats_latest_arr_pos = 0;
+	}
+
+	total_packets = stats_overall.iso_recv_count + stats_overall.iso_lost_count;
+
+	if ((total_packets % 100) == 0) {
```

Comment:
This is intentional. We log for each 100 of total packets (this may work poorly for very low or very high intervals), but that number is unrelated to the last 1000 packets. 

It is in theory possible to create a formula to make both of these numbers be relative to the number of CIS and the interval, but that's a thing for the future if ever.